File: 05 Support Vector Regression/svr.py
# -*- coding: utf-8 -*-
"""
Created on Sat Jan  4 22:37:49 2020

@author: Daniel
"""
# SVR

# Plantilla de Pre Procesado

# Cómo importar las librerías
import numpy as np
import matplotlib.pyplot as plt
import pandas as pd

# Importar el data set
dataset = pd.read_csv('Position_Salaries.csv')
X = dataset.iloc[:, 1:2].values
y = dataset.iloc[:, 2].values


# Dividir el data set en conjunto de entrenamiento y conjunto de testing
"""
from sklearn.model_selection import train_test_split
X_train, X_test, y_train, y_test = train_test_split(X, y, test_size = 0.2, random_state = 0)
"""

# Escalado de variables
from sklearn.preprocessing import StandardScaler
sc_X = StandardScaler()
sc_y = StandardScaler()
X = sc_X.fit_transform(X)
y = sc_y.fit_transform(y.reshape(-1, 1))

#Ajustar la regresión con el dataset
from sklearn.svm import SVR
regression  = SVR(kernel ="rbf")
regression.fit(X,y)

#Predicción de nuestros modelos con SVR
value = np.array(6.5).reshape(-1,1)
y_pred = sc_y.inverse_transform(regression.predict(sc_X.transform(value)))

# Visualización de los resultados del SVR
plt.scatter(X, y, color = "red")
plt.plot(X, regression.predict(X), color = "blue")
plt.title("Modelo de regresión (SVR)")
plt.xlabel("Posición de Empleado")
plt.xlabel("Sueldo (en $)")
plt.show()

File: 05 Support Vector Regression/SVR.R
# SVR

# Importar el dataset

dataset = read.csv('05 Support Vector Regression/Position_Salaries.csv')
dataset = dataset[, 2:3]

#Ajustar SVR con el conjunto de datos
#install.packages("e1071")
library(e1071)
regression = svm(formula = Salary ~ .,
                 data = dataset,
                 type = "eps-regression",
                 kernel = "radial")

#Prediccion de nuevos resultadops con SVR 
y_pred = predict(regression, newdata = data.frame(Level= 6.5))

# Visualización del modelo SVR
#install.packages("ggplot2")
library(ggplot2)
x_grid = seq(min(dataset$Level), max(dataset$Level), 0.1)
ggplot() +
  geom_point(aes(x = dataset$Level , y = dataset$Salary),
             color = "red") +
  geom_line(aes(x = x_grid, y = predict(regression, 
                                        newdata = data.frame(Level = x_grid))),
            color = "blue") +
  ggtitle("Predicción (Modelo de Regresión)") +
  xlab("Nivel del empleado") +
  ylab("Sueldo (en $)")
